include/ceed/types.h

```diff
@@ -13,6 +13,16 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#ifndef __NO_INLINE__
```

Comment:
I think the discussion in this PR is great. It might be a good idea to put some of that into a comment like we have for the `CeedPragmaSIMD` macro.

Also, this might be worth including in a stand alone PR.

Reply:
A separate PR makes sense to me. Working on that now.